slurm/slurm.py

```diff
@@ -131,25 +138,142 @@ def assert_affinity(self):
+}
+EOF
+"""
+
+    @sanity_function
+    def assert_partitions(self):
+        with open('partitions.json', 'r', encoding='utf-8') as json_file:
+            partitions = json.load(json_file)
+
+        return sn.all([
+            sn.assert_not_found(".", self.stderr, self.descr + ': no error messages'),
+            sn.assert_eq(
+                {'skylake', 'broadwell', 'skylake_mpi', 'ivybridge_mpi'},
+                set(partitions['singlenode'].split(',')),
+                self.descr + ': singlenode partitions expected: {0}, found: {1}'
+            ),
+            sn.assert_eq(
+                {'skylake_mpi', 'ivybridge_mpi'},
+                set(partitions['multinode'].split(',')),
+                self.descr + ': multinode partitions expected: {0}, found: {1}'
+            ),
+            sn.assert_eq(
+                {'skylake_mpi', 'ivybridge_mpi'},
+                set(partitions['manycores'].split(',')),
+                self.descr + ': manycores partitions expected: {0}, found: {1}'
+            ),
+            sn.assert_eq(
+                {'ampere_gpu', 'pascal_gpu'},
+                set(partitions['gpunode'].split(',')),
+                self.descr + ': gpunode partitions expected: {0}, found: {1}'
+            ),
+        ])
+
+
+@rfm.simple_test
+class WarningGPU(SlurmTestBase):
+    descr += ": warning for gpu options"
+    tags.add('local')
+    executable = tempjob.format('--gpus=1') + tempjob.format('-n 1 --gpus-per-task=1')
+
+    @sanity_function
+    def assert_warning(self):
+        return sn.all([
+            sn.assert_found(
+                r'WARNING:\S\[0m GPU option \S\[1;31m--gpus\S\[0m is not recommended for performance',
+                self.stderr,
+                self.descr + ': --gpus'
+            ),
+            sn.assert_found(
+                r'WARNING:\S\[0m GPU option \S\[1;31m--gpus-per-task\S\[0m is not recommended for performance',
+                self.stderr,
+                self.descr + ': --gpus-per-task'
+            ),
+            sn.assert_eq(
+                sn.count(sn.extractall(r'^job submitted: (?P<jobid>\S+)', self.stdout, 'jobid')),
+                2,
+                self.descr + ": 2 jobs submitted successfully"
+            )
+        ])
+
+
+@rfm.simple_test
+class WarningMultiGPU(SlurmTestBase):
+    descr += ": warning multi-GPU jobs without --ntasks-per-gpu"
+    tags.add('local')
+    executable = tempjob.format('--gpus-per-node=2')
+
+    @sanity_function
+    def assert_warning(self):
+        return sn.all([
+            sn.assert_found(
+                r'Please use \S\[1;32m--ntasks-per-gpu\S\[0m and \S\[1;32m--gpus-per-node',
+                self.stderr,
+                self.descr
+            ),
+            sn.assert_found(r"^job submitted: \d+$", self.stdout, self.descr + ": job submitted successfully"),
+        ])
+
+
+@rfm.simple_test
+class NonGPUInGPUPartition(SlurmTestBase):
+    # this test will fail for users with account bvo00005
```

Comment:
not sure how to make this work

Reply:
can we add it as a test that is not run by default?

Reply:
added a check that will skip the test for bvo00005